includes/functions.php

```diff
@@ -3963,4 +3963,84 @@ function pmpro_get_ip() {
 	$client_ip = preg_replace( '/[^0-9a-fA-F:., ]/', '', $client_ip );
 	
 	return $client_ip;
+}
+
+/**
+ * Determines if this order can be refunded
+ * @param  object $order The order that we want to refund
+ * @return bool Returns a bool value based on if the order can be refunded
+ */
+function pmpro_allowed_refunds( $order ) { 
+
+	if( empty( $order ) || empty( $order->gateway ) ) {
+		return false;
+	}
+
+	/**
+	 * Processes a refund for Stripe orders
+	 *
+	 * @since TBA
+	 *
+	 * @param bool Default return value is false to determine if the refund was successfully processed. 
+	 * @param object $order The Member Order we want to refund
+	 * @return bool If the refund was successfully processed
+	 */
+	
+	$disallowed_statuses = apply_filters( 'pmpro_disallowed_refund_statuses', array( 'pending', 'refunded' ) );
+	
+	if( 
+		!in_array( $order->status, $disallowed_statuses, true ) && //Don't allow pending orders to be refunded
+		in_array( $order->gateway, array( 'stripe', 'paypalexpress' ), true )  //Only apply to these gateways
+	) {
+		return true;
+	}
+
+	return false;
+}
+
+
+/**
+ * Decides which filter should be used for the refund depending on gateway
+ * @param  object $order Member Order that we are refunding
+ * @return bool 	Returns a bool value based on if a refund was processed successfully or not
+ */
+function pmpro_refund_order( $order ){
+
+	if( empty( $order ) ){
+		return;
+	}
+
+	//Not going to refund an order that has already been refunded
+	if( $order->status == 'refunded' ) {
+		return true; 
+	}
+
+	if( $order->gateway === 'stripe' ) {
+		/**
+		 * Processes a refund for Stripe orders
+		 *
+		 * @since TBA
+		 *
+		 * @param bool Default return value is false to determine if the refund was successfully processed. 
+		 * @param object $order The Member Order we want to refund
+		 * @return bool If the refund was successfully processed
```

Comment:
```suggestion
		 * @since TBD
		 *
		 * @param bool $success Default return value is false to determine if the refund was successfully processed. 
		 * @param object $order The Member Order we want to refund.
```